infrastructure/code/settings.go

```diff
@@ -17,11 +17,68 @@ package code
 
 import (
 	"net/url"
+	"os"
+	"strconv"
 	"strings"
+	"sync"
 
 	"github.com/snyk/snyk-ls/application/config"
+	"github.com/snyk/snyk-ls/internal/concurrency"
 )
 
+const (
+	// String "true" or "false", see `defaultAutofixEnabled`
+	autofixEnabledEnvVarKey = "SNYK_AUTOFIX_ENABLED"
+	// By default this is disabled
+	defaultAutofixEnabled = false
+)
+
+var (
+	codeSettingsSingletone      *codeSettings
```

Comment:
Any reason to have this on a "global" package level? Why not to make it part of the code scanner singleton?

Reply:
https://github.com/snyk/snyk-ls/pull/234#discussion_r1131187779 
> moving config as a property of the scanner was less preferrable because the codeaction creation happens at the level of bundle, which has no reverse access to the scanner that created this bundle

maybe I'm wrong here, you mention that scanner is the singleton but I AFAIU accesing it goes via `snyk` package, then going to its product scanners, filtering them to find the code one etc -- just to arrive back to the code.

As opposed to that chain, keeping it just a config singleton at a package level allows to access it now (and in future) without overhead. Do I oversimplify or miss smth?